backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/highlight/impl/ElasticsearchHighlightConfiguration.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.search.highlight.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Set;
+
+import org.hibernate.search.backend.elasticsearch.logging.impl.Log;
+import org.hibernate.search.backend.elasticsearch.search.common.impl.ElasticsearchSearchIndexScope;
+import org.hibernate.search.engine.search.highlight.HighlightConfiguration;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+import com.google.gson.JsonObject;
+
+public interface ElasticsearchHighlightConfiguration extends HighlightConfiguration {
+
+	Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
+	void request(JsonObject requestBody);
+
+	Set<String> indexNames();
+
+	static ElasticsearchHighlightConfiguration from(ElasticsearchSearchIndexScope<?> scope,
+			HighlightConfiguration configuration) {
+		if ( !( configuration instanceof ElasticsearchHighlightConfiguration ) ) {
+			throw log.cannotMixElasticsearchSearchQueryWithOtherHighlightConfigurations( configuration );
+		}
+		ElasticsearchHighlightConfiguration casted = (ElasticsearchHighlightConfiguration) configuration;
+		if ( !scope.hibernateSearchIndexNames().equals( casted.indexNames() ) ) {
+			throw log.highlightConfigurationDefinedOnDifferentIndexes( configuration, casted.indexNames(),
+					scope.hibernateSearchIndexNames()
+			);
+		}
+		return casted;
+	}
+
+	boolean isConfigured();
```

Comment:
I'm not sure how configuration could be unconfigured... maybe you need a better name for this?